play/src/front/Chat/Connection/Matrix/MatrixChatConnection.ts

```diff
@@ -253,23 +553,21 @@ export class MatrixChatConnection implements ChatConnectionInterface {
 
             const room = this.client.getRoom(room_id);
             if (!room) return;
-            const newRoom = new MatrixChatRoom(room);
-            this.roomList.set(room_id, newRoom);
-            return newRoom;
+            return this.createAndAddNewRootRoom(room);
         } catch (error) {
             throw this.handleMatrixError(error);
         }
     }
 
     getDirectRoomFor(userID: string): ChatRoom | undefined {
         const directRooms = Array.from(this.roomList)
-            .filter(([_, room]) => {
+            .filter(([, room]) => {
```

Comment:
I think you can use `this.roomList.values()` that returns an array of only the values of a map.